samples/subsys/usb/cdc_acm_composite/src/main.c

```diff
@@ -28,47 +28,50 @@ LOG_MODULE_REGISTER(cdc_acm_composite, LOG_LEVEL_INF);
 uint8_t buffer0[RING_BUF_SIZE];
 uint8_t buffer1[RING_BUF_SIZE];
 
-static struct serial_data {
-	const struct device *peer;
-	struct serial_data *peer_data;
-	struct ring_buf ringbuf;
-} peers[2];
+struct serial_peer {
+	const struct device *dev;
+	struct serial_peer *data;
+	struct ring_buf rb;
+};
+
+#define DEFINE_SERIAL_PEER(node_id) { .dev = DEVICE_DT_GET(node_id), },
+static struct serial_peer peers[] = {
+	DT_FOREACH_STATUS_OKAY(zephyr_cdc_acm_uart, DEFINE_SERIAL_PEER)
+};
+
+BUILD_ASSERT(ARRAY_SIZE(peers) >= 2, "Not enough CDC ACM instances");
```

Comment:
Shouldn't this be asserted to be equal 2?

Reply:
No, because there can be a node defined at the board level, e.g. for console, this example requires at least two to build.